src/main/java/Duke.java

```diff
@@ -1,10 +1,212 @@
+        String description = parsedCommand[0];
+        String by = parsedCommand[1];
+        Deadline t = new Deadline(description, by);
+        tasks[numTasks] = t;
+        numTasks += 1;
+    }
+
+    public static void executeEvent(String input) {
+        String[] parsedCommand = parseDeadlineOrEvent(input);
+        String description = parsedCommand[0];
+        String at = parsedCommand[1];
+        Event t = new Event(description, at);
+        tasks[numTasks] = t;
+        numTasks += 1;
+    }
+
+    //Returns a String array with the first element as the description  and the second element as the deadline/time period
+    public static String[] parseDeadlineOrEvent(String input) {
+        String command = getCommand(input);
+        String[] inputArray;
+        if (command.equals("deadline")) {
+            inputArray = input.split(" /by ");
+        } else {
+            inputArray = input.split(" /at ");
+        }
+        inputArray[0] = inputArray[0].substring(inputArray[0].indexOf(" "));
+        return inputArray;
+    }
+
+    public static void listTasks() {
+        if (numTasks == 0) {
+            System.out.println("You don't seem to have any tasks so far!");
+        } else {
+            for (int i = 0; i < numTasks; i++) {
+                System.out.println((i + 1) + ". " + tasks[i]);
+            }
+        }
+    }
+
+    public static String getCommand(String input) {
+        String[] arrayOfInput = input.split(" ", 0);
+        return arrayOfInput[0];
+    }
+
+    public static String getDescription(String input) {
+        return input.substring(input.indexOf(" "));
+    }
+
+    public static String getInput() {
+        System.out.println("Type a valid command (bye, list, mark, unmark, todo, deadline, event) or add a task to your list:");
+        Scanner scan = new Scanner(System.in);
+        return scan.nextLine();
+    }
+
+    public static int getInteger() {
+        Scanner scan = new Scanner(System.in);
+        return scan.nextInt();
+    }
+
+    public static boolean isValidCommand(String command) {
+        return isList(command) || isMarkAsDone(command) || isUnmarkAsDone(command) || isTodo(command)
+                || isDeadline(command) || isEvent(command);
+    }
+
+    public static boolean isBye(String command) {
+        return command.equals("bye");
+    }
+
+    public static boolean isList(String command) {
+        return command.equals("list");
+    }
+
+    public static boolean isMarkAsDone(String command) {
+        return command.equals("mark");
+    }
+
+    public static boolean isUnmarkAsDone(String command) {
+        return command.equals("unmark");
+    }
+
+    public static boolean isTodo(String command) {
+        return command.equals("todo");
+    }
+
+    public static boolean isDeadline(String command) {
+        return command.equals("deadline");
+    }
+
+    public static boolean isEvent(String command) {
+        return command.equals("event");
+    }
```

Comment:
May not need all these methods if you use a default case as invalid command in the executeCommand's switch statement?

Reply:
Will definitely consider this idea!